### ISSUE TITLE
Clearlist specific CWs (content warnings I'm not concerned about)

### ISSUE THREAD

**Issue by ryanprior** [mastodon/mastodon#18219]
### Pitch

I propose that we allow a user to clearlist an exact CW string like (eg "food") which will show posts by default if and only if they match that exact CW in its entirety. i.e. "CW food" would match, but "CW food/diet" would not match; you'd clearlist that CW separately if desirable.

### Motivation

I often see posts marked "CW: politics" or "CW: food" which I'm fine with. These constitute a majority of the CWs I see on Mastodon. I understand why people use them and support people in using CWs boldly: it's a basically pro-social pattern. But they mostly don't improve my experience any and I'm always clicking through one after another. However, I don't want to show all content by default, because things marked "CW: gore" I never want to see at all, and a "CW: transphobia" I might want to see or not depending on the day.

If I could clearlist exact matches, it might take me a while to catch the variants and typos, but this keeps the feature simple to implement and to understand while allowing my experience to incrementally improve over time as I clearlist more CWs that don't bother me.

Related issues:
- Some folks in https://github.com/mastodon/mastodon/issues/1123 mentioned a whitelist, but @trwnh noted that whitelists could get complicated. This proposal keeps it simple by operating on exact matches.
- In https://github.com/mastodon/mastodon/issues/8681 we see a proposal for CW categories, such that we can decide per-category whether we want to show those posts or not. Commenters rightly point out that choosing the categories would be tricky and that localizing and synchronizing them would present further challenges. Operating on exact matches essentially turns each CW string into its own category: an intentionally less powerful implementation in this case is easier to understand and skirts a lot of issues.
- The proposal in https://github.com/mastodon/mastodon/issues/14268 is to allow auto-expanding CWs within a thread. This proposal overlaps in the common case that you're expanding all the CWs because you don't care about them at all, although there's another case where it's a CW you do care about in general but want to expand the posts in a single thread.

**Comment by BenAveling** [comment]
Suggestion: As well as "Show more", have an option for "Always show content with this CW".

**Comment by gringer** [comment]
Another related suggestion: auto-complete prompts for CW strings, to reduce the number of variants and typos that would cause problems with whitelists.

**Comment by eishiya** [comment]
People often list multiple warnings in varying orders in the CW, so I don't think an exact match would be very effective for a lot of people. However, some sort of whitelist or blacklist for auto-expanding CWs and auto-showing media would be a fantastic feature. Exact CW matches should be an option, but not the only option.

It would also be interesting to go the opposite route - automatically "CW" posts with certain keywords, so that people with less common triggers don't have to be entirely reliant on other people to tag them. It would make a nice third option for the Filters feature - instead of dropping the toot entirely or showing it as "Filtered", it could show "Toot contains: <matched filter(s)>" and let the user click it to view the toot if they wish.

**Comment by ryanprior** [comment]
@eishiya agreed on all points, thank you for adding! You can use [this link](https://github.com/mastodon/mastodon/discussions/new) to bring up a topic for discussion. You can link back here and it'll add a note to this thread.

**Comment by leahccinnamon** [comment]
I strongly agree on this issue. I am a new user, and content warnings on pornography was a welcome feature. As an artist, it was hardly possible on twitter to check someone who followed you or replied to you without stumbling over something you don't quite want to see. Hence I was surprised to see content warnings used mostly on things you would think would not require them for a broader public, such as food or eye contact.

Personally, I dislike that a large amount of posts are filtered, especially in regards to politics and racial discussion. It can come to tone policing if talking about racism is something that should always be hidden behind a read more.

Also, it seems to not be uniformly enforced, today I scrolled past my local timeline, saw several content warnings for food, several food images without content warnings and at least two penises without any filter attached to them 🙈

Personally, I would like to see the food, as images of colorful food make me happy, but not see any genitals without consenting to it. So the possibility to opt in to -specific- content warnings always being expanded, such as politics and food, would be a great help to let me filter it to my personal happy experience. Another option would be for certain tags or keywords to always be hidden by a content warning for a user, and the user to be able to set those filters similar to hashtag filters.

**Comment by oligneisti** [comment]
I think the OP has this well figured out.

For example (inspired by the replies on this topic):

Someone posts a picture of their penis in a hotdog-bun. This person could then put (#)DickPic in the CW and it wouldn't matter if he jokingly included (#)food in either the text or CW.

Of course this could be abused but no more so than the current system. It is then up to us admins to take reports of such behavior seriously.
